src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -0,0 +1,124 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+import re
+from knack.util import CLIError
+
+from ._utils import is_valid_connection_string, resolve_resource_group
+from .azconfig.models import QueryFields
+
+
+def validate_tags(namespace):
+    if namespace.tags is not None:
+        if not namespace.tags:
+            return
+        tag_content = namespace.tags.split(',')
+        for tag_kv in tag_content:
+            tag = tag_kv.split("=")
+            if len(tag) != 2:
+                raise CLIError(
+                    "Tags are not valid. It should be this format 'key1=value1,key2=value2'")
+
+
+def validate_datetime(namespace):
+    ''' valid datetime format:YYYY-MM-DDThh:mm:ssZ '''
+    datetime_format = '^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])T(2[0-3]|[01][0-9]):[0-5][0-9]:[0-5][0-9][a-zA-Z]{0,5}$'
+    if namespace.datetime is not None and re.search(datetime_format, namespace.datetime) is None:
+        raise CLIError(
+            'The input datetime is invalid. Correct format should be YYYY-MM-DDThh:mm:ssZ ')
+
+
+def validate_connection_string(namespace):
+    ''' Endpoint=https://example.azconfig.io;Id=xxxxx;Secret=xxxx'''
+    connection_string = namespace.connection_string
+    if connection_string:
+        if not is_valid_connection_string(connection_string):
+            raise CLIError('''The connection string is invalid. \
+Correct format should be Endpoint=https://example.azconfig.io;Id=xxxxx;Secret=xxxx ''')
+
+
+def validate_import_depth(namespace):
+    depth = namespace.depth
+    if depth is not None:
+        try:
+            depth = int(depth)
+            if depth < 1:
+                raise CLIError('Depth should be at least 1.')
+        except ValueError:
+            raise CLIError("Depth is not a number.")
+
+
+def validate_separator(namespace):
+    if namespace.separator is not None:
+        if namespace.format_ == "properties":
+            raise CLIError("Separator is not needed for properties file.")
+        valid_separators = ['.', ',', ';', '-', '_', '__', '/', ':']
+        if namespace.separator not in valid_separators:
+            raise CLIError(
+                "Unsupported separator, allowed values: '.', ',', ';', '-', '_', '__', '/', ':'.")
+
+
+def validate_import(namespace):
+    source = namespace.source
+    if source == 'file':
+        if namespace.path is None:
+            raise CLIError("Please specify the file path.")
+        if namespace.format_ is None:
+            raise CLIError("Please specify the file format.")
```

Comment:
Errors should favor usage-string format. Also, it is better to figure out ALL of the missing parameters in a parameter set and present one message. With this approach the first call can fail on the file path and then the next call on the file format. A useful error message will ensure the second call succeeds because the user will know to supply both.
Example: `usage error: --path PATH --format FORMAT`